### ISSUE TITLE
Data generation 

### ISSUE THREAD

**Issue by rruizdeaustri** [timothygebhard/magic-bullet#5]
Dear authors,

I'm trying to generate data to reproduce your results in the paper and to produce 
my own results using alternative ML models. For that I'm using ggwd 
and I have doubts:

1. In the paper you say that training data is composed of ~ 30k 
examples, each containing 8 sec with a sampling rate of  ~2k Hz 
(after down sampling the original data). With these numbers 
the total number of samples for training is ~ 30k*2k*8 
Is this correct ?

2. The examples are got taking randomly GPS times 
with ggwd (in the case of signals a waveform is injected on top).
Then for training, did you just add all of them from the hdf5 
file produced by ggwd (the 30k examples) ? 

If so, is it not an issue that in this way the time in the 
training set is not continuous ?

Thanks a lot !!

All the best,
Roberto

**Comment by timothygebhard** [comment]
Dear Roberto,

thanks for your interest in our work, and for reaching out to us! :) Unfortunately, we are not quite sure if we understand your questions correctly, but here is our best attempt at answering them:

1. This is really just about what one means by "sample" / "example", isn't it? Our training data set consists of 32768 _time series_ (which we also call "examples"), each with two channels (H1/L1) of length 8s and with a sampling rate of 2048 Hz, so that indeed comes out at 2^15 * 2 * 2^3 * 2^11 = 2^30 "time steps" / numbers (personally, I wouldn't call these "samples").
2. We are not quite sure what you mean with "the time in the training set is not continuous". The times at which we sample the noise from real data are drawn randomly and uniformly from the "admissible" parts of O1 (admissible = good data quality, no real events, ...). In `ggwd`, this sampling is done by `NoiseTimeline.sample()`. Now of course, as the noise times are drawn independently, it might be the case that there are parts of O1 that got used multiple times as noise, and other times did not get used at all (after all, O1 lasted longer than just 32k * 8s ≈ 3 days), but we do not believe this to be an issue. The goal is only to present the network with enough representative real noise for learning; but since O1 (and its duration) is not special per se, we do not see a need to use all of it. In fact, as we mention in section VII.C of the paper, our final network was also able to find the events from O2, even though it was not specially trained for that.

We hope that this helps you / answers your questions! :)

All the best,
— Timothy

**Edit:** Added channels (H1/L1).

**Comment by rruizdeaustri** [comment]
Dear Timothy,


Thanks a lot for the clarification !!

I have a couple of more questions:

1. You train L1 and H1 data simultaneously, right  ?
If so half of the time series is from H1 and the another half 
from L1 (16384). Correct ?

2. In practice how do you label the time series ? 
  I mean, if they have injections then you set it to 
  one number and noise ones to another ?

Thanks,
Rbt

**Comment by timothygebhard** [comment]
Dear Roberto,

1. No, our examples consist of time series that have two channels (for H1 and L1). This is because in case of a signal, the time series for H1 and L1 are not independent, and the time delay between the signals in the two channels, the relative signal amplitudes, etc. all depend on the location of the source in the sky, so in order to capture these correlations correctly, we generate the data for H1 and L1 jointly. I had left out this bit in my previous answer in an attempt not to add more confusion, but I've just edited my answer. Long story short: We have 32k examples, and each example consists of two time series ("channels"), one for H1 and one for L1.
2. We describe the creation of the labels (and our motivation for our specific approach) in detail in section VI.A of the paper.

Cheers,
— Timothy
